task6/task.js

```diff
@@ -0,0 +1,287 @@
+   	likes: ['max', 'Иванов Иван']
+	},
+	{
+		id: '12',
+   	description: 'Hello there',
+   	createdAt: new Date('2020-03-19T23:00:00'),
+   	author: 'Иванов Иван',
+   	photoLink: 'images/1.png',
+   	hashTags: ['#gelo','#ni'],
+   	likes: ['max', 'Иванов Иван']
+	},
+	{
+		id: '13',
+   	description: 'фЫврш гффы фы тв ргц фыв',
+   	createdAt: new Date('2000-03-20T23:00:00'),
+   	author: 'Да да я',
+   	photoLink: 'images/1.png',
+   	hashTags: ['#gelo','#ni'],
+   	likes: ['max', 'Иванов Иван']
+	},
+	{
+		id: '14',
+   	description: 'фЫврш гффы фы тв ргц фыв',
+   	createdAt: new Date('2020-03-20T23:00:00'),
+   	author: 'Да да я',
+   	photoLink: 'images/1.png',
+   	hashTags: ['#gelo','#ni'],
+   	likes: ['max', 'Иванов Иван']
+	},
+	{
+		id: '15',
+   	description: 'фЫврш гффы фы тв ргц фыв',
+   	createdAt: new Date('2020-03-21T23:00:00'),
+   	author: 'Макс Борис',
+   	photoLink: 'images/1.png',
+		hashTags: ['#gelo','#ni'],
+   	likes: ['max', 'Иванов Иван']
+	},
+	{
+		id: '16',
+   	description: 'АХАХАХАХАХАХА',
+   	createdAt: new Date('2020-03-21T23:00:00'),
+   	author: 'Да да я',
+   	photoLink: 'images/1.png',
+   	hashTags: ['#gelo','#ni'],
+   	likes: ['max', 'Иванов Иван']
+	},
+	{
+		id: '17',
+   	description: 'фЫврш гффы фы тв ргц фыв',
+   	createdAt: new Date('2020-03-22T23:00:00'),
+   	author: 'Да да я',
+   	photoLink: 'images/1.png',
+   	hashTags: ['#gelo','#ni'],
+   	likes: ['max', 'Иванов Иван']
+	},
+	{
+		id: '18',
+   	description: 'фЫврш гффы фы тв ргц фыв',
+   	createdAt: new Date('2020-03-23T23:00:00'),
+   	author: 'Да да я',
+   	photoLink: 'images/1.png',
+   	hashTags: ['#gelo','#ni'],
+   	likes: ['max', 'Иванов Иван']
+	},
+	{
+		id: '19',
+   	description: 'фЫврш гффы фы тв ргц фыв',
+   	createdAt: new Date('2020-03-24T23:00:00'),
+   	author: 'Да да я',
+   	photoLink: 'images/1.png',
+   	hashTags: ['#gelo','#ni'],
+   	likes: ['max', 'Иванов Иван']
+	},
+	{
+		id: '20',
+   	description: 'фЫврш гффы фы тв ргц фыв',
+   	createdAt: new Date('2020-03-25T23:00:00'),
+   	author: 'Да да я',
+   	photoLink: 'images/1.png',
+   	hashTags: ['#gelo','#ni'],
+   	likes: ['max', 'Иванов Иван']	
+	},
+];
+let func = (function() {
+	function comp(a, b) {
+		return a.createdAt - b.createdAt;
+	}
+	return {
+		getPosts: function(skip = 0, top = 10, filterConfig ={}) {
+			let newPosts = [];
+			let filterKeys = Object.keys(filterConfig);
+			let filt;
+			if(filterKeys.length != 0)
+			{
+				filt = posts.filter(post => {
+					if(filterKeys[0] != 'hashTags')
+						return post[filterKeys[0]] == filterConfig[filterKeys[0]]
+					for(tag of filterConfig.hashTags)
+						if(post.hashTags.indexOf(tag) >= 0)
+							return true;
+					return false;
+				 });
+			}
+			else
+				filt = posts;
+			filt.sort(comp);
+			for (let i = skip; i < Math.min(skip + top, filt.length) ; i++) {
+				newPosts.push(filt[i]);
+			}
+			return newPosts;
+		},
+		getPost: function(id = '1'){
+			for(post of posts)
```

Comment:
Can here `find` method be used? https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Array/find